### ISSUE TITLE
Dead connection message received delay in stream management

### ISSUE THREAD

**Issue by Jethvasagar** [esl/MongooseIM#1583]
MongooseIM version: (2.0.0)
Installed from: (source)
Erlang/OTP version: (19)

I have enable stream management in both side client or server. I have two users A and B. Both users are online.Then user A suddenly lose his connection. but A user still appear online on user B and as well as on server. During that time user B sending message on user A. Those message are not lost but when user A is appear online again it will receive those message after 2-3 minute.and i will get message stanza on Offline storage and delivery receipt i will got on SM storage.This issue same occur on one to one chat and mucLight. have i need to customized any mongooseIM modules. Please guide me why users received delay message when their are lost his connection. is it possible to changed SM storage to offline storage(MAM). here's link for same issue i have fetch (https://www.ejabberd.im/faq/tcp) but  have not lost my messages but just received it delay.

**Comment by deadjdona** [comment]
maybe you should update to latest version of MIM?

**Comment by Jethvasagar** [comment]
My latest version of MIM is 2.0

**Comment by Jethvasagar** [comment]
Following stanza i will receive when i get delay message
`<message to='sagardj@host.com' from='4e051e98-64f0-43b3-8698-a74361edbf1e@muclight.host.com/demo@host.com' id='85ED708A-5928-4C70-A968-F925E1B4F7A7' type='groupchat'><body>Hi</body><request xmlns='urn:xmpp:receipts'/><delay xmlns='urn:xmpp:delay' stamp='2017-12-01T10:33:48.065+00:00' from='localhost'>SM Storage</delay></message>`

**Comment by Jethvasagar** [comment]
Is it possible to fetch SM storage messages same like MAM archive messages ?

**Comment by fenek** [comment]
Hi @Jethvasagar 

Let me rephrase what you've written to ensure that I understand you problem well.

You are asking why MongooseIM doesn't send buffered messages right after reconnecting, but waits 2-3 minutes instead, correct? And you are using new stream ID or resuming the session with existing ID? What client do you use?

**Comment by Jethvasagar** [comment]
yes @fenek you are correct. i am using smack-4.2 client. i am resuming the session with existing ID.but if i create new connection with smack its used new stream id and previous session all messages coming late.i have already set `stream management resumption` is true in my smack lib `xmppTcpConnection.setUseStreamManagement(true); xmppTcpConnection.setUseStreamManagementResumption(true);`

**Comment by fenek** [comment]
I've checked Smack source code and it seems that stream ID is stored in private variable in TCP connection object. I suppose you have to reuse this object for reconnection. Are you doing it or creating new connection object when trying to resume?

**Comment by Jethvasagar** [comment]
I create a new connection object when app start it first time. connection trying to resume I have reuse this object for reconnection.

**Comment by fenek** [comment]
Hi @Jethvasagar 

First of all, I apologise for late response. I didn't have time to reply earlier. :( Is this issue still valid for you?

**Comment by fenek** [comment]
Closing due to inactivity. Please reopen if the issue persists.
